src/linker/Linker.Dataflow/ValueNode.cs

```diff
@@ -385,7 +385,7 @@ public static bool DetectCycle (this ValueNode node, HashSet<ValueNode> seenNode
 
 				case ValueNodeKind.LoadField:
 					LoadFieldValue lfv = (LoadFieldValue)node;
-					foundCycle = lfv.InstanceValue.DetectCycle (seenNodes, allNodesSeen);
+					foundCycle = lfv.InstanceValue != null && lfv.InstanceValue.DetectCycle (seenNodes, allNodesSeen);
```

Comment:
We could potentially also just delete the `InstanceValue` property - we don't model these so they will pretty much always be null. It's a leftover from .NET Native I probably should have deleted.